python/TestHarness/TestHarness.py

```diff
@@ -358,11 +358,11 @@ def findAndRunTests(self, find_only=False):
         if self.options.input_file_name != '':
             self._infiles = self.options.input_file_name.split(',')
 
-        if self.options.spec_file and os.path.isdir(self.options.spec_file):
-            search_dir = self.options.spec_file
-        elif self.options.spec_file and os.path.isfile(self.options.spec_file):
-            search_dir = os.path.dirname(self.options.spec_file)
-            self._infiles = [os.path.basename(self.options.spec_file)]
+        if self.options.run_dir and os.path.isdir(self.options.run_dir):
+            search_dir = self.options.run_dir
+        elif self.options.run_dir and os.path.isfile(self.options.run_dir):
```

Comment:
This is the combination that I believe needs to go. We don't want to accept a file in "--run_dir". You can put the error message right here!

Reply:
This was done, thank you for the feedback!